src/higher-level/index-hl.c

```diff
@@ -136,6 +136,107 @@ float tmp, ind, scale = 10000.0;
 }
 
 
+/** This function computes a spectral index time series, with a Ratio - 1,
++++ e.g. CIre: (b1/b2)-1
+--- ard:    ARD
+--- mask_:  mask image
+--- ts:     pointer to instantly useable TSA image arrays
+--- b1:     band 1
+--- b2:     band 2
+--- nc:     number of cells
+--- nt:     number of ARD products over time
+--- nodata: nodata value
++++ Return: void
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++**/
+void index_ratio_minus1(ard_t *ard, small *mask_, tsa_t *ts, int b1, int b2, int nc, int nt, short nodata){
+int p, t;
+float ind, scale = 10000.0;
+
+
+  #pragma omp parallel private(t,ind) shared(ard,mask_,ts,b1,b2,nc,nt,nodata,scale) default(none)
+  {
+
+    #pragma omp for
+    for (p=0; p<nc; p++){
+
+      if (mask_ != NULL && !mask_[p]){
+        for (t=0; t<nt; t++) ts->tss_[t][p] = nodata;
+        continue;
+      }
+
+      for (t=0; t<nt; t++){
+
+        if (!ard[t].msk[p]){
+          ts->tss_[t][p] = nodata;
+        } else {
+          ind = (ard[t].dat[b1][p]/ard[t].dat[b2][p])-1;
+          if (b2 == 0){
```

Comment:
--> ard[t].dat[b2][p]

Reply:
I believe that integer under/overflow is possible with this equation. 

if you do not have a valid range (like in NDVI), I suggest to use something like:

if (ard[t].dat[b2][p] == 0 || ind*scale > SHRT_MAX || ind*scale < SHRT_MIN)){

Reply:
> if (b2 == 0){

This is a typical case to be covered by a unittest. Syntax is fine but it is obviously a bug which would avoid any nodata in the output.

Reply:
> if (ard[t].dat[b2][p] == 0 || ind*scale > SHRT_MAX || ind*scale < SHRT_MIN)){

Implemented in https://github.com/davidfrantz/force/pull/90/commits/62d9e4fed1b65ce65dad5bd2486975183deab8e9.

Reply:
not necessarily, if b2 is the blue band, this would always be nodata ;)

Reply:
But b2 is is only an integer index, right?